math/src/main/java/com/powsybl/math/timeseries/TimeSeries.java

```diff
@@ -9,29 +9,284 @@
+import com.google.common.primitives.Doubles;
 import com.powsybl.commons.json.JsonUtil;
+import gnu.trove.list.array.TDoubleArrayList;
+import org.slf4j.LoggerFactory;
+import org.threeten.extra.Interval;
 
-import java.io.IOException;
-import java.io.UncheckedIOException;
-import java.io.Writer;
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
 import java.nio.file.Path;
-import java.util.ArrayList;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Objects;
+import java.time.Duration;
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
 import java.util.stream.Stream;
 
 /**
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  */
-public interface TimeSeries<P extends AbstractPoint> extends Iterable<P> {
+public interface TimeSeries<P extends AbstractPoint, T extends TimeSeries<P, T>> extends Iterable<P> {
 
     TimeSeriesMetadata getMetadata();
 
     Stream<P> stream();
 
     Iterator<P> iterator();
 
+    List<T> split(int n);
+
+    static <P extends AbstractPoint, T extends TimeSeries<P, T>> List<List<T>> split(List<T> timeSeriesList, int n) {
+        Objects.requireNonNull(timeSeriesList);
+        List<List<T>> splitList = new ArrayList<>(n);
+        for (int i = 0; i < n; i++) {
+            splitList.add(new ArrayList<>(timeSeriesList.size()));
+        }
+        for (T timeSeries : timeSeriesList) {
+            List<T> split = timeSeries.split(n);
+            for (int i = 0; i < n; i++) {
+                splitList.get(i).add(split.get(i));
+            }
+        }
+        return splitList;
+    }
+
+    static Map<Integer, List<TimeSeries>> parseCsv(Path file) {
+        return parseCsv(file, TimeSeriesConstants.DEFAULT_SEPARATOR);
+    }
+
+    static Map<Integer, List<TimeSeries>> parseCsv(String csv, char separator) {
+        try (BufferedReader reader = new BufferedReader(new StringReader(csv))) {
+            return parseCsv(reader, separator);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    static Map<Integer, List<TimeSeries>> parseCsv(Path file, char separator) {
+        try (BufferedReader reader = Files.newBufferedReader(file, StandardCharsets.UTF_8)) {
+            return parseCsv(reader, separator);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    static double parseDouble(String token) {
+        return token.isEmpty() ? Double.NaN : Double.parseDouble(token);
+    }
+
+    static String checkString(String token) {
+        return token.isEmpty() ? null : token;
+    }
+
+    class CsvParsingContext {
+        final List<String> names;
+
+        final TimeSeriesDataType[] dataTypes;
+        final Object[] values;
+
+        final List<ZonedDateTime> times = new ArrayList<>();
+
+        TimeSeriesIndex refIndex;
+
+        CsvParsingContext(List<String> names) {
```

Comment:
Could names be null?

Reply:
No but this is an internal class than is not part of the API